### PR TITLE
2024-01-01-preview

### PR DESCRIPTION
**Pull request by jamesfan1**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py [hunk 1]
```diff
@@ -201,10 +213,45 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
+            required=True,
+        )
+        _element.name = AAZStrArg(
+            options=["name"],
+            required=True,
+        )
+
+        capacity = cls._args_schema.scaling_properties.profiles.Element.capacity
+        capacity.max = AAZIntArg(
+            options=["max"],
```
Comment (puneetsarna) [inline src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py:239]: I am not sure how this looks like on Azure CLI but "min" and "max" are not good names as they don't tell the user much about what the min and max are about or applied to.

File: src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py [hunk 2]
```diff
@@ -143,6 +148,13 @@ def _build_arguments_schema(cls, *args, **kwargs):
             help={"short-summary": "Optional: Preferred communication email", "long-summary": "Usage --user-profile preferred-email=xyz@abc.com"},
         )
 
+        auto_upgrade_profile = cls._args_schema.auto_upgrade_profile
+        auto_upgrade_profile.upgrade_channel = AAZStrArg(
```
Comment (puneetsarna) [inline src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py:152]: is there a way to add some "help" around valid values? If so, can you please add `preview` and `stable`

File: src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py (hunk 1 above)
Comment (jamesfan1) [reply src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py:239]: [{name:default,capacity:{min:10,max:30}}]

File: src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py (hunk 2 above)
Comment (jamesfan1) [reply src/nginx/azext_nginx/aaz/latest/nginx/deployment/_create.py:152]: fixed

**Comment by jamesfan1** [comment]
tests are passing locally, not sure why it doesnt pass in the pipeline

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by AllyW** [comment]
> ## ⚠️ Suggestions
> ### Module: nginx
> * Update version to `2.0.0b1` in setup.py
> 
> #### Notes
> * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

@jamesfan1 there is a breaking change in this pr which deleted a parameter. If it's unintentional, please fix it, if it is intented, that's ok.

Another problem is the version number, the last released version is `1.0.0`, since there is a breaking change, so the next release version would be `2.0.0`. And if you plan to release a preview version, it should be `2.0.0b1`.   `1.0.0b1` is smaller than `1.0.0` and we should not release an older version than current one.

For version details, please refer to our doc [here](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md)

File: src/nginx/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+1.0.0b1
```
Comment (AllyW) [inline src/nginx/HISTORY.rst:6]: please correct version number

Comment (jamesfan1) [reply src/nginx/HISTORY.rst]: fixed, thank you

**Comment by jamesfan1** [comment]
@AllyW
 
image

I am unsure what is wrong for 
Azure.azure-cli-extensions Integration test
 
it is passing locally, and generated a recording file
 
however, on ci it's failing with
 
ERROR    cli.azure.cli.core.azclierror:azlogging.py:212 Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/nginx/azext_nginx/tests/latest/recordings/test_deployment_cert_config.yaml') in your current record mode ('once').
 
on this line, but this line runs fine locally
        public_ip = self.cmd('network public-ip create --resource-group {rg} --name {public_ip_name} --version IPv4 --sku Standard --zone 2').get_output_in_json()
 
 
have you seen such error?

**Review by AllyW — APPROVED** [review]

File: src/nginx/azext_nginx/azext_metadata.json [hunk 4]
```diff
@@ -1,3 +1,3 @@
 {
-    "azext.minCliCoreVersion": "2.55.0"
+    "azext.minCliCoreVersion": "2.58.0"
```
Comment (jsntcy) [inline src/nginx/azext_nginx/azext_metadata.json:2]: Please set `azext.isPreview` to true as you will release it as preview.

**Review by jsntcy — APPROVED** [review]

**Comment by jamesfan1** [comment]
added the tag, please merge when possible. Thank you all for the help!!
@jsntcy @AllyW

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**